### PR TITLE
feat(admin): makes it easier to navigate plugin dependencies

### PR DESCRIPTION
**Pull request by hypeJunction**
Improves admin plugin management interface by listing plugin dependents
and conflicts with shortcut URLs. Modifies some styles and views to
make colors less aggressive, texts more readable and anchors recognizable.
Adds ElggPlugin::canDeactivate()

admin-plugin

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -71,3 +71,4 @@ startup.sh
 !.gitignore
 !.scripts
 !.tx
+/bower_components/
```
Comment (jdalsem) [inline .gitignore]: needed for this PR? or something from your own installation

File: views/default/object/plugin/invalid.php [hunk 2]
```diff
@@ -9,35 +9,40 @@
...
-		<div class="pts">
-			<?php
-				echo elgg_view('output/url', array(
-					'href' => "#elgg-plugin-manifest-$css_id",
-					'text' => elgg_echo("admin:plugins:label:moreinfo"),
-					'rel' => 'toggle',
-				));
-			?>
+<div class="elgg-state-draggable elgg-plugin elgg-state-inactive elgg-state-cannot-activate" id="<?php echo $css_id; ?>">
+	<div class="elgg-image-block">
```
Comment (jdalsem) [inline views/default/object/plugin/invalid.php]: can't we use elgg_view_image_block?

**Review by jdalsem — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (hypeJunction) [inline .gitignore]: Looks like some artifact left when switching branches. I will remove.

File: views/default/object/plugin/invalid.php (hunk 2 above)
Comment (hypeJunction) [inline views/default/object/plugin/invalid.php]: I suppose we can.

**Review by mrclay — APPROVED** [review]

File: engine/classes/ElggPlugin.php [hunk 3]
```diff
@@ -744,6 +748,52 @@ public function activate($site_guid = null) {
...
+		if (!empty($dependents)) {
+			$list = array_map(function(\ElggPlugin $plugin) {
+				$css_id = preg_replace('/[^a-z0-9-]/i', '-', $plugin->getManifest()->getID());
+				return elgg_view('output/url', [
+					'text' => $plugin->getManifest()->getName(),
+					'href' => "#$css_id",
+				]);
+			}, $dependents);
+			$name = $this->getManifest()->getName();
+			$list = implode($list);
```
Comment (hypeJunction) [inline engine/classes/ElggPlugin.php]: Meh

File: engine/classes/ElggPlugin.php [hunk 4]
```diff
@@ -744,6 +748,52 @@ public function activate($site_guid = null) {
...
+				if ($required['type'] == 'plugin' && $required['name'] == $this->getID()) {
+					// there are active dependents
+					$dependents[$manifest->getPluginID()] = $plugin;
+				}
+			}
+		}
+
+		if (!empty($dependents)) {
+			$list = array_map(function(\ElggPlugin $plugin) {
+				$css_id = preg_replace('/[^a-z0-9-]/i', '-', $plugin->getManifest()->getID());
```
Comment (hypeJunction) [inline engine/classes/ElggPlugin.php]: We use this often. Let's add API

File: engine/classes/ElggPluginPackage.php [hunk 5]
```diff
@@ -403,7 +403,12 @@ public function checkDependencies($full_report = false) {
 					$conflict['name'] = $plugin->getManifest()->getName();
 
 					if (!$full_report && !$result['status']) {
-						$this->errorMsg = "Conflicts with plugin \"{$plugin->getManifest()->getName()}\".";
+						$css_id = preg_replace('/[^a-z0-9-]/i', '-', $plugin->getManifest()->getID());
+						$link = elgg_view('output/url', [
+							'text' => $plugin->getManifest()->getName(),
+							'href' => "#$css_id",
+						]);
+						$this->errorMsg = "Conflicts with plugin: $link";
```
Comment (hypeJunction) [inline engine/classes/ElggPluginPackage.php]: Add translation

**Comment by mrclay** [comment]
I agree with your reviewing :)

**Comment by mrclay** [comment]
Cannot deactivate plugins should have a white background, as they're active. And similarly cannot activate plugins should have the light grey background.

**Comment by mrclay** [comment]
Idea: Instead of "Cannot activate" on the button, use "Activate <ban icon>" http://fontawesome.io/icon/ban/ and just take the regular button color and give the whole button opacity:.75. Similarly for Cannot deactivate. I can do this in a follow-up PR.

**Comment by hypeJunction** [comment]
Not fan of ban icon. Prefer button with disabled atate

**Pull request merged**